Sources/Bot/Services/GitHubService.swift

```diff
@@ -1,29 +1,36 @@
 import Foundation
 import ReactiveSwift
 import Result
+import CryptoSwift
 
 public final class GitHubService {
 
     internal enum HTTPHeader: String {
         case event = "X-GitHub-Event"
+        case signature = "X-Hub-Signature"
     }
 
     internal enum APIEvent: String {
         case pullRequest = "pull_request"
         case ping
     }
 
-    private let scheduler: QueueScheduler
+    public typealias Token = String
+
+    private let scheduler: Scheduler
+    private let signatureVerifier: (RequestProtocol) -> Result<RequestProtocol, EventHandlingError>
```

Comment:
Is this still required to have that injected as a closure now that we don't inject a signatureVerifier in tests?

Might as well remove the closure property here and use the private method directly

Reply:
I don't want to keep the token accessible in memory, capturing it in the closure makes it harder to be accessed although as everything not impossible.

Reply:
Ah, good point 👍